src/main/kotlin/de/bund/digitalservice/useid/timebasedtokens/TBTMockDatasource.kt

```diff
@@ -0,0 +1,28 @@
+package de.bund.digitalservice.useid.timebasedtokens
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Mono
+import java.time.LocalDateTime
+import java.util.UUID
+
+@Repository
+@ConditionalOnProperty(name = ["features.desktop-solution-enabled"], havingValue = "true")
+class TBTMockDatasource {
+    private val timeBasedTokens = mutableListOf<TimeBasedToken>()
+
+    fun save(timeBasedToken: TimeBasedToken): Mono<TimeBasedToken> {
+        timeBasedToken.createdAt = LocalDateTime.now()
+        timeBasedTokens.add(timeBasedToken)
+        return Mono.just(timeBasedToken)
+    }
+
+    fun deleteAllByUseIdSessionId(useIdSessionId: UUID): Mono<Void> {
+        timeBasedTokens.removeAll { it.useIdSessionId == useIdSessionId }
+        return Mono.empty()
+    }
+
+    fun findByUseIdSessionIdAndTokenId(useIdSessionId: UUID, tokenId: UUID): Mono<TimeBasedToken?> {
```

Comment:
Why is there a `?` in the return type? The time based token will never be null right? Either a tbt is found and returned wrapped by `Mono`, or an empty `Mono` is returned. Would make the code of the caller easier.